dpnp/backend/custom_kernels_statistics.cpp

```diff
@@ -288,17 +300,28 @@ void custom_min_c(void* array1_in, void* result1, const size_t* shape, size_t nd
     {
         size *= shape[i];
     }
+    if constexpr (std::is_same<_DataType, double>::value || std::is_same<_DataType, float>::value)
+    {
+        // Required initializing the result before call the function
+        result[0] = array_1[0];
+
+        // https://docs.oneapi.com/versions/latest/onemkl/mkl-stats-make_dataset.html
+        auto dataset = mkl_stats::make_dataset<mkl_stats::layout::row_major>(1, size, array_1);
 
-    auto policy = oneapi::dpl::execution::make_device_policy<class custom_min_c_kernel<_DataType>>(DPNP_QUEUE);
+        // https://docs.oneapi.com/versions/latest/onemkl/mkl-stats-min.html
+        cl::sycl::event event = mkl_stats::min(DPNP_QUEUE, dataset, result);
 
-    _DataType* res = std::min_element(policy, array_1, array_1 + size);
-    policy.queue().wait();
+        event.wait();
+    }
+    else
+    {
+        auto policy = oneapi::dpl::execution::make_device_policy<class custom_min_c_kernel<_DataType>>(DPNP_QUEUE);
 
-    result[0] = *res;
+        _DataType* res = std::min_element(policy, array_1, array_1 + size);
+        policy.queue().wait();
```

Comment:
It looks like you `wait()` for whole `queue` instead wafting for particular `event`.

Reply:
Yes, i think so too. But at this moment we don't know the way how to get single event from PSTL call.

Reply:
I didn't find anything useful about PSTL calls synchronization to fix that somehow.